src/app/components/History/index.tsx

```diff
@@ -28,6 +28,16 @@ function History(): JSX.Element | null {
     [fingerprint, removeFromHistory],
   );
 
+  const handleOpenPreview = useCallback(
+    (url: string) => {
+      if (!isOpen) {
+        setUrl(url);
+        open();
+      }
+    },
+    [isOpen, open, close],
+  );
```

Comment:
```suggestion
  const handleOpenPreview = useCallback(
    (url: string) => {
      open(url);
    },
    [open],
  );
```

Dans l'idée c'est mieux. On appelle juste une méthode pour ouvrir et c'est elle qui va set correctement l'url ainsi que check si c'est déjà ouvert ou pas au lieu de déporter la logique dans des composants qui techniquement, n'en ont rien à faire